### PR TITLE
Make delete safer

### PR DESCRIPTION
**Pull request by gregorjerse**
Validate arguments to the delete methed and make sure connector can not
perform operation outsde  its base_path.

Also fix cleanup bug that could potentially delete data belonging to
other storage locations and make sure StorageLocation is not deleted
if its data was not successfully removed.

### CODE REVIEW

File: docs/storage.rst [hunk 1]
```diff
@@ -6,6 +6,17 @@ Resolwe Storage Framework is storage management system for Resolwe Flow Design.
 Currenlty it supports storing data to local filesystem, Google Cloud Storage
 and Amazon Simple Storage Service.
 
+
+Testing connectors
```
Comment (dblenkus) [inline docs/storage.rst:10]: Move documentation update to a separate commit as it is not connected to the browsable view.

File: resolwe/storage/views.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                    "type": "directory",
+                    "mtime": self._get_mtime(file_storage.created),
+                }
+                for file_storage in FileStorage.objects.filter(
+                    storage_locations__status=StorageLocation.STATUS_DONE
+                ).distinct()
+            ]
+            return JsonResponse(out, safe=False)
+
+        data_qset = Data.objects.filter(pk=data_id)
```
[the remark this answers is not in the document]
Comment (gregorjerse) [reply resolwe/storage/views.py:81]: OK.

File: resolwe/storage/views.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        :param request: the given request. When the user requests URL
+            http://platform-centos.genialis.local/data/1/..., then the value of
+            parameter should be 1/... .
+        """
+
+        data_id = kwargs.get("data_id")
+        relative_path = Path(kwargs.get("uri", ""))
+
+        # List all FileStorage objects.
+        if data_id is None:
```
[the remark this answers is not in the document]
Comment (gregorjerse) [reply resolwe/storage/views.py:68]: I will remove this part of the code.

File: docs/storage.rst (hunk 1 above)
Comment (gregorjerse) [reply docs/storage.rst:10]: OK.
